### ISSUE TITLE
Release of pre-trained weights?

### ISSUE THREAD

**Issue by eware-godaddy** [Alibaba-MIIL/ML_Decoder#3]
Thanks so much for this amazing new model. 

I was wondering if you would be able to release some of the pre-trained weights behind these core benchmark datasets?

I pointed some colleagues to your paper and repo to show them what the SOTA was in this area, and couldn't easily point them to a collab notebook, or huggingface demo, or even build our own hosted demo to show them.

Love your work, please keep up the great work!

**Comment by mrT23** [comment]
Have you reviewed the Readme file ?

we provide reproduction code for all the main results of the paper (single-label, multi-label, ZSL).

we can surely release pre-trained weights, but letting people validate our results via actual reproducible code is better, we think

**Comment by eware-godaddy** [comment]
Hi, yes I've read the README and your paper too. I love how you got rid of the self-attention layer. That was was a big problem with the query2label model.

I guess there are multiple users of your work? 

There are fellow researchers who are trying to replicate your work, and build on your ideas, and what you've provided for reproducibility is fantastic.

However, there is another class of customer of your work who would just like to pick and up and try to use your work directly in their applications.

So, asking them to download the entire COCO dataset and train a model is a significant barrier. (At least hours of work for someone experienced).

I do note there are ZSL options as well to get people started. But even providing some weights will enable that community and users to build collab notebooks, hosted demos to showcase your excellent work, and upload it to sites like huggingface, etc, as well as increase the visibility of your work.

Anyway, if you were able to share this in some way that would very much appreciated by not only my team, but I'm sure thousands of others who would like to quickly evaluate your work, or use it directly in a ZSL context.

Amazing work again!

**Comment by mrT23** [comment]
ok, i am convinced :-)

will share some model pre-train weights next week

**Comment by eware-godaddy** [comment]
Thank you so much 🙇 ! I really appreciate it!

**Comment by mrT23** [comment]
i have added example models and inference and validation code

enjoy :)

**Comment by eware-godaddy** [comment]
Thank you. This is amazing!
